Automattic/pocket-casts-ios
# Adds a way to export the database and preferences from the Help & Feedback page

This adds a button that will export and allow sharing of a zip file that contains the podcast database and UserDefaults.

The process to share the file is:
1. Creating a temporary directory
2. Copying the database into it
3. Copying the preferences file into it
4. Using the NSFileCoordinator to generate a zip
5. Saving the resulting zip to the temporary directory
6. Sharing it
7. Deleting the temporary files after sharing

**Note:** When sharing the system shows some warnings, but doesn't stop us sharing the file. I tried a lot of things to get this to stop, but none worked. 

## Demo Video


https://github.com/Automattic/pocket-casts-ios/assets/793774/f0a722ab-df08-4e20-9ddb-afdad425d1b5





## To test
1. Launch the app
2. Go to the Profile tab
3. Tap on Help & Feedback
4. Tap the Share button in the top corner
5. ✅ You should see the exporting message
6. ✅ After a few seconds you should see the share sheet with the file name "Pocket Casts Export"
7. Share the file to your Mac
8. Open it
9. ✅ Verify it unzips
10. ✅ Verify it contains both the database and preferences files
11. ✅ Verify the database is valid
12. ✅ Verify the preferences are valid

## Checklist
